graspologic/pipeline/embed/adjacency_spectral_embedding.py

```diff
@@ -0,0 +1,209 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+from typing import Optional, Union
+import warnings
+
+import networkx as nx
+import numpy as np
+
+from . import __SVD_SOLVER_TYPES  # from the module init
+from ._elbow import _index_of_elbow
+from .embeddings import Embeddings
+from ..preconditions import (
+    check_argument,
+    check_argument_types,
+    check_optional_argument_types,
+    is_real_weighted,
+)
+from graspologic.embed import AdjacencySpectralEmbed
+from graspologic.utils import is_fully_connected, pass_to_ranks
+
+
+def adjacency_spectral_embedding(
+    graph: Union[nx.Graph, nx.DiGraph],
+    dimensions: int = 100,
+    elbow_cut: Optional[int] = None,
+    svd_solver_algorithm: str = "randomized",
+    svd_solver_iterations: int = 5,
+    svd_seed: Optional[int] = None,
+    weight_attribute: str = "weight",
+) -> Embeddings:
+    """
+    Given a weighted directed or undirected networkx graph (*not* multigraph),
+    generate an Embeddings object.
+
+    Adjacency spectral embeddings are extremely egocentric.
+
+    `Adjacency Spectral Embedding Tutorial
+    <https://microsoft.github.io/graspologic/tutorials/embedding/AdjacencySpectralEmbed.html>`_
+
+    In addition to diagonal augmentation, all graphs will have pass to ranks executed.
+
+    Parameters
+    ----------
+    graph : Union[nx.Graph, nx.DiGraph]
+        A simple, weighted graph, either undirected or directed. The graph **must** be
+        fully weighted (every edge has a weight), it **must** be a simple graph (meaning
```

Comment:
simple graph may be an overloaded term here https://mathworld.wolfram.com/SimpleGraph.html#:~:text=A%20simple%20graph%2C%20also%20called,edges%20(Gibbons%201985%2C%20p.&text=A%20simple%20graph%20may%20be,refers%20to%20a%20simple%20graph.

Reply:
what's the short and sweet descriptor of a graph that is explicitly not a multigraph?

Reply:
i guess I don't know of one other than "not multigraph"

Reply:
ended up going with basic graph, and if that has a term overload then I don't know.  describing something as "not something more complex" vexes me